org.opentosca.planbuilder.type.plugin.ubuntuvm/src/org/opentosca/planbuilder/type/plugin/ubuntuvm/handler/AbstractUbuntuVmHandler.java

```diff
@@ -0,0 +1,241 @@
+
+public abstract class AbstractUbuntuVmHandler implements UbuntuVmHandler {
+
+	private static final Logger LOG = LoggerFactory.getLogger(AbstractUbuntuVmHandler.class);
+
+	abstract protected boolean handleWithLocalCloudProviderInterface(AbstractNodeTemplate abstractNodeTemplate);
+
+	abstract protected boolean handleWithCloudProviderInterface(AbstractNodeTemplate abstractNodeTemplate);
+
+	abstract protected boolean handleGeneric(AbstractNodeTemplate abstractNodeTemplate);
+
+	abstract protected boolean handleWithDockerEngineInterface(AbstractNodeTemplate nodeTemplate);
+
+	@Override
+	public boolean handle(AbstractNodeTemplate nodeTemplate) {
+		if (nodeTemplate == null) {
+			return false;
+		}
+
+		LOG.debug("Checking if nodeTemplate " + nodeTemplate.getId() + " can be handled");
+
+		// cloudprovider node is handled by doing nothing
+		if (org.opentosca.container.core.tosca.convention.Utils
+				.isSupportedCloudProviderNodeType(nodeTemplate.getType().getId())) {
+			return true;
+		}
+
+		// docker engine node is handled by doing nothing
+		if (org.opentosca.container.core.tosca.convention.Utils
+				.isSupportedDockerEngineNodeType(nodeTemplate.getType().getId())) {
+			return true;
+		}
+
+		// when infrastructure node arrives start handling
+		if (org.opentosca.container.core.tosca.convention.Utils
+				.isSupportedInfrastructureNodeType(nodeTemplate.getType().getId())) {
+			// check if this node is connected to a cloud provider node type, if true ->
+			// append code
+			for (final AbstractRelationshipTemplate relation : nodeTemplate.getOutgoingRelations()) {
+				if (org.opentosca.container.core.tosca.convention.Utils
+						.isSupportedCloudProviderNodeType(relation.getTarget().getType().getId())) {
+					if (relation.getTarget().getType().getId().equals(Types.openStackLiberty12NodeType)
+							| relation.getTarget().getType().getId().equals(Types.vmWareVsphere55NodeType)
+							| relation.getTarget().getType().getId().equals(Types.amazonEc2NodeType)) {
+						// bit hacky now, but until the nodeType cleanup is
+						// finished this should be enough right now
+						return this.handleWithCloudProviderInterface(nodeTemplate);
+					} else if (relation.getTarget().getType().getId().equals(Types.localHypervisor)) {
+						return this.handleWithLocalCloudProviderInterface(nodeTemplate);
+					} else {
+						return this.handleGeneric(nodeTemplate);
+					}
+				}
+			}
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * This method checks whether the given QName represents a Ubuntu NodeType which
+	 * has an implicit Ubuntu Image (e.g. Ubuntu 13.10 Server)
+	 *
+	 * @param nodeType
+	 *            a QName
+	 * @return true if the given QName represents an Ubuntu NodeType with implicit
+	 *         image information
+	 */
+	protected boolean isUbuntuNodeTypeWithImplicitImage(final QName nodeType) {
+		if (this.createUbuntuImageStringFromNodeType(nodeType) == null) {
+			return false;
+		} else {
+			return true;
+		}
```

Comment:
can be simplified to `return this.createUbuntuImageStringFromNodeType(nodeType) != null`